services/federator/test/unit/Test/Federator/ExternalServer.hs

```diff
@@ -72,59 +71,79 @@ exampleRequest certFile path = do
         trBody = "\"foo\""
       }
 
+data Call = Call
+  { cComponent :: Component,
+    cPath :: ByteString,
+    cBody :: LByteString,
+    cDomain :: Domain
+  }
+  deriving (Eq, Show)
+
+mockService ::
+  Members [Output Call, Embed IO] r =>
```

Comment:
How come this works without using a value-level list? I was expecting a type-level list here.

Reply:
It's not a value level list. The promotion tick is optional when the type is unambiguous. For the case of lists, this means that `[A, B]` is interpreted as something of kind `[*]` (i.e. the same as `'[A, B]`, aka `A ': B ': '[]`), but `[A]` is interpreted as something of type `*` (i.e. the type of lists of `A`).

We can turn on `-Wunticked-promoted-constructors` to turn this into an error, if desired. Should we?